beacon_node/lighthouse_network/gossipsub/src/behaviour.rs

```diff
@@ -1828,8 +1825,12 @@ where
                 peer_score.duplicated_message(propagation_source, &msg_id, &message.topic);
             }
             self.mcache.observe_duplicate(&msg_id, propagation_source);
+
+            // Broadcast IDONTWANT messages.
+            self.send_idontwant(&raw_message, &msg_id, propagation_source);
```

Comment:
I think we want this after the return, after the if statement. 

At the moment, this will broadcast an IDONTWANT only if the message is a duplicate. 

We want the logic to broadcast only if the message isn't a duplicate.

Reply:
oh yeah makes sense!